mlflow/tracing/trace_manager.py

```diff
@@ -63,64 +61,21 @@ def __init__(self):
         self._trace_id_to_request_id = {}
         self._lock = threading.Lock()  # Lock for _traces
 
-    def start_detached_span(
```

Comment:
note: Moved this method from the trace manager to the provider module to separate responsibility more clearly.
* trace manager - a simple registry stores spans and traces, doesn't create or modify those entities.
* provider - contains functions to creates spans and traces.

Reply:
Isn't the role of TracerProvider to provide a tracer, and the tracer is responsible for creating spans and traces? I'd expect the tracer to do the creation.

Reply:
Maybe the following helps: I noticed that OTel defines a concept of an IdGenerator (https://opentelemetry-python.readthedocs.io/en/latest/sdk/trace.id_generator.html#opentelemetry.sdk.trace.id_generator.IdGenerator), which can be passed to the TraceProvider constructor. Instead of adding methods to the provider module for constructing traceinfos, perhaps all we need to do is implement an IdGenerator that calls `MlflowClient().create_trace_info()` when `generate_trace_id()` is invoked?

Reply:
I think one problem in that approach is the responsibility of `create_trace_info()` is not only getting request ID, but other backend-generated info like `mlflow.artifactLocation` stored in the returned TraceInfo. This is why the trace manager needs to keep track of `TraceInfo` not only `request_id`.

We could make `create_trace_info()` or `generate_trace_id()` register the returned TraceInfo into the trace manager, but it makes the IDGenerator responsibility a bit over its intended scope and make MLflow and OTel more entangled - currently the span creation is pretty much unidirectional like Otel (create span) -> MLflow (wrap it), but it will be like MLflow -> OTel (id generator)-> MLflow (client/backend) -> OTel (actual span creation from id) -> MLflow (wrap it).